### ISSUE TITLE
[PROPOSAL] Time and Lighting improvements

### ISSUE THREAD

**Issue by FAStre** [AoTTG-2/AoTTG-2#576]
**How will this enhancement improve the game?**
#113 introduced a new time and lighting system for AoTTG2. The feature is stable and looks great, but there are some improvements and suggestions for further improving it.

**Describe the solution you'd like**

-  The ability to change lighting quality settings, and editing it independent of general graphics settings, wait for #536 before adding.

-  Moon phases and having some nights be without moonlight(this should be done only after torches/other light sources are added to the game, because night would be very dark without moonlight).

-  Revisiting light and darkness levels, at the moment the lighting is set to have quite dark shadows but bright moonlight. Players can move quite comfortably while in full moonlight but darkness is a hamper on being able to move as freely as one would want. This is intentional as #113's issue description mentioned making players seek out light sources, which for now is just moonlight, until light sources are introduced into the game.

- Being able to control Day Length as well as Night Length(being able to have longer sunlight and vice versa)

- More optimization(using a static lightmap while the cycle is paused? There are some [tools](https://github.com/laurenth-personal/lightmap-switching-tool)
that allow one to switch between static and real time lighting at run time)

- Titan AI changes

**Additional context** (Remove if not relevant)
There are improvements that could come to #113 in the branch itself but as it stands now it is a huge pull request, and to minimize future merge conflicts it's best to just merge the stable feature and do significant upgrades and improvements in a separate issue.

**Contributor instructions** (Do not modify)
1. Determine if the enhancement is compatible with existing features
2. If the idea is denied, please provide a reason and close the issue. If the idea is accepted, consult your team to do an estimation of effort. Remove the [PROPOSAL] name and label.
3. Move the issue to the Future Milestone. On rare exceptions an issue may be moved to Release if there's sufficient upvotes.
4. Once the issue matches the current milestone, follow development procedures.

**Comment by atirutw** [comment]
> * The ability to change lighting quality settings, and editing it independent of general graphics settings, wait for #536 before adding.

I'm not sure if this can be done but I'll do some research.

> * Moon phases and having some nights be without moonlight(this should be done only after torches/other light sources are added to the game, because night would be very dark without moonlight).
> * Being able to control Day Length as well as Night Length(being able to have longer sunlight and vice versa)

We'll probably want to make a more complex & complete day/night cycle solution similar to [Azure[Sky]](https://assetstore.unity.com/packages/tools/particles-effects/azure-sky-dynamic-skybox-36050) with all sort of features we could want but the architecture will have to change, mainly separating the lighting logic from the networking(maybe splitting them into different scripts) to make things easier to implement.

As for the implementation, I'd like to add the ability to transition between timecycle profiles(will be done in the lighting logic) in anticipation for the weather system, event system similar to Azure's(On Second/Minute/Hour change, etc.) for titan AI, and maybe some more stuff I'm not aware we'll want yet.
